svg lettericons

gurgeous opened the pull request:
The server can generate SVG lettericons. Only enabled if `?formats=` includes `svg`. Most of the action is in `RenderSVG`. Also tweaks to `IconPath` to support SVG. For example, an SVG lettericon path doesn't include the size in the filename.

One thing to note. The A-Z lettericons look great (see below), but ф is a bit low. Right now we have `dy="0.10em"` hardcoded in the SVG template. Not sure if it would be possible/worthwhile to customize based on the letter. I could also imagine using Freetype to measure the Arial glyph height... The traco 1M doesn't include any IDN domains, unfortunately.

![image](https://user-images.githubusercontent.com/526195/109394978-c817ae00-78de-11eb-939d-01a7b5d90a5f.png)

mat commented on besticon/iconserver/server_test.go:
```diff
@@ -188,6 +201,21 @@ func TestGetLetterIcon(t *testing.T) {
 }
 
+func TestGetLetterIconSVG(t *testing.T) {
+	req, err := http.NewRequest("GET", "/lettericons/M-144-EFC25D.svg", nil)
```
Do we even want to allow for the `size` to be in the path for `.svg`? I'm thinking the fact the we allow it is kind of ok by Postel's law on the one hand but could lead to misunderstanding one the other (we're not using this).

gurgeous replied:
Sure, let's turn off support for size entirely. We're not generating those with SVG so no need to parse them. Give me a sec here

mat commented on lettericon/lettericon.go:
```diff
@@ -281,6 +304,45 @@ func percentDecode(p string) string {
+	}
+
+	// vars
+	vars := map[string]string{
+		"$BG_COLOR": ColorToHex(bgColor),
+		"$FG_COLOR": ColorToHex(pickForegroundColor(bgColor)),
+		"$LETTER":   buf.String(),
```
Like the lightweight templating approach for this👍

mat reviewed (approved):
Well done, looks really great, let's tackle improvement of the non A-Z's later, this is a great starting point, thank you 👍 Really just one comment not the svg lettericon paths.

gurgeous commented:
OK, check out the latest commit. I reworked `ParseIconPath` a bit to consume params one at a time. Also added some tests for bad paths to make sure things were working properly. Thanks mat! 😄
